socorro/unittest/processor/test_processor_app.py

```diff
@@ -277,6 +277,48 @@ def mocked_save_raw_and_processed(*_):
         assert captured_exception.__class__ == ValueError
         assert captured_exception.args[0] == 'Someone is wrong on the Internet'
 
+    @mock.patch('socorro.lib.raven_client.raven')
+    def test_transform_get_error_with_raven_configured_successful(self, mock_raven):
+        # First mock everything so we're not sending real things anywhere
+        captured_exceptions = []  # a global
+
+        def mock_capture_exception(exc_info=None):
+            captured_exceptions.append(exc_info)
+            return 'someidentifier'
+
+        raven_mock_client = mock.MagicMock()
+        raven_mock_client.captureException.side_effect = mock_capture_exception
+
+        mock_raven.Client.return_value = raven_mock_client
+
+        # Set up a processor and mock .get_raw_crash() in that
+        config = self.get_standard_config(
+            sentry_dsn='https://abc123@example.com/project'
+        )
+        pa = ProcessorApp(config)
+        pa._setup_source_and_destination()
+
+        def mocked_get_raw_crash(*_):
+            raise ValueError('simulated error')
+
+        pa.source.get_raw_crash.side_effect = (
```

Comment:
IIRC if we just set `side_effect` to the exception itself, it will throw it when called automatically.

Reply:
We look at the exception message to make sure it's the exception we threw here rather than some other exception thrown elsewhere. I think we need a function for that. If not, can you show an example of what you're thinking about?

Reply:
Just do an equality check against the exception you assign to `side_effect`:

```python
>>> import mock
>>> mock_thing = mock.Mock()
>>> error = ValueError('foo')
>>> mock_thing.foo.side_effect = error
>>> try:
...     mock_thing.foo()
...     assert false
... except ValueError as err:
...     assert err == error
```

Or, in this specific case, assign the same exception to `side_effect` that you pass to `assert_called_once_with`.